### ISSUE TITLE
Case should not be used to opt out (filter) candidates

### ISSUE THREAD

**Issue by fniessen** [senny/emacs-eclim#186]
I'll try to describe a problem that my colleague (whom I'm trying to convert from Eclipse to Emacs) gets with completion.

Let's suppose we have the following vars in a Java file:
    ┌────
    │ UnreadableVar
    │ CamelCase123
    │ CamelcaseABC
    └────

Now, I will try to be as accurate as possible what my colleague has in
mind...

1. When he types "Unreadable"...

   He expects to see the Company menu displaying "UnreadableVar", right?

   He expects that typing either "v" or "V" will correctly make the
   completion to "UnreadableVar", right?

   This isn't the case today with Company-mode -- and it seems to be the
   responsability of the Company backend.

2. When he types "Camel"...

   He expects to see the Company menu displaying both "CamelCase123" and
   "CamelcaseABC", right?

   He expects that typing "c" or "C" won't help him discriminate between
   both strings.

   He expects to be able to complete the common part (that is,
   "CamelCase" or "Camelcase" -- depending on what he wrote) by using
   `company-complete-common'.

In summary, for him, the case should not be used to filter within the
list of candidates.

**Comment by kleewho** [comment]
OmniSharp somehow is enable to make `company-mode` case sensitive or insensitive. [This line](https://github.com/OmniSharp/omnisharp-emacs#company-mode-interface)  handles this so there's a way in `company-mode`. Probably I should start from looking at the `company` sources.

**Comment by dgutov** [comment]
@kleewho It should be enough for you to just look at the `company-backends` docstring. Specifically, the `ignore-case` command.

**Comment by kleewho** [comment]
@dgutov yes, you're right. I tend to look at `ensime` and `omnisharp` cause those projects are very similar to what `emacs-eclim` wants to achieve and I almost by accident found the line with case sensitivity
